open-hackathon-server/src/hackathon/views/schema/AdminHostserverResource

```diff
@@ -0,0 +1,143 @@
+        "properties": {
+        }
+      }
+    },
+
+    "post": {
+      "input": {
+        "title": "request to create a new hackathon",
+        "description": "args to create new hackathon. Args that not listed here are NOT acceptable.",
+        "type": "object",
+        "properties": {
+          "id": {
+            "description": "id of the host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "hackathon_id": {
+            "description": "id of the hackathon",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "vm_name": {
+            "description": "the name of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "public_dns": {
+            "description": "the public_dns of vm/host_server",
+            "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
+            "required": true
+          },
+          "public_ip": {
+            "description": "the public_ip of vm/host_server",
+            "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
+            "required": true
+          },
+          "public_docker_api_port": {
+            "description": "the public_docker_api_port of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "private_ip": {
+            "description": "the private ip of vm/host_server",
+            "pattern": "^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$",
+            "required": true
+          },
+          "private_docker_api_port": {
+            "description": "the private_docker_api_port of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "container_count": {
+            "description": "the container count of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "container_max_count": {
+            "description": "the max container count of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          }
+        }
+      }
+    },
+
+    "put": {
+      "input": {
+        "title": "request to update the information of a host server",
+        "description": "args to update the information of a host server",
+        "type": "object",
+        "properties": {
+          "id": {
+            "description": "id of the host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "hackathon_id": {
+            "description": "id of the hackathon",
+            "pattern": "^[0-9]+$",
+            "required": true
+          },
+          "vm_name": {
+            "description": "the name of vm/host_server",
+            "pattern": "^[0-9]+$",
+            "required": true
```

Comment:
And for `put` method, we shouldn't ask frontend caller for all fields. For example, if `vm_name` is not included in http body, that means the caller don't want to update field `vm_name` so we leave as it is.  `"required": true` is not suitable for those cases. Same for the next validations.